pints/_abc/_abc_rejection.py

```diff
@@ -0,0 +1,85 @@
+#
+# ABC Rejection method
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+
+
+class RejectionABC(pints.ABCSampler):
+    """
+    Implements the rejection ABC algorithm as described in [1].
+
+    Here is a high-level description of the algorithm:
+
+        theta* ~ p(theta), i.e. sample parameters from prior distribution
+        x ~ p(x|theta*), i.e. sample data from sampling distribution
+        if s(x) < threshold, theta* added to list of samples
+
+    References
+    ----------
+    .. [1] "Approximate Bayesian Computation (ABC) in practice". Katalin
+           Csillery, Michael G.B.Blum, Oscar E. Gaggiotti, Olivier Francois
+           (2010) Trends in Ecology & Evolution
+           https://doi.org/10.1016/j.tree.2010.04.001
+
+    """
+    def __init__(self, log_prior):
+
+        self._log_prior = log_prior
+        self._threshold = 1
+        self._xs = None
+        self._ready_for_tell = False
+
+    def name(self):
+        """ See :meth:`pints.ABCSampler.name()`. """
+        return 'Rejection ABC'
+
+    def ask(self, n_samples):
+        """ See :meth:`ABCSampler.ask()`. """
+        if self._ready_for_tell:
+            raise RuntimeError('Ask called before tell.')
+        self._xs = self._log_prior.sample(n_samples)
+
+        self._ready_for_tell = True
+        return self._xs
+
+    def tell(self, fx):
+        """ See :meth:`ABCSampler.tell()`. """
+        if not self._ready_for_tell:
+            raise RuntimeError('Tell called before ask.')
+        self._ready_for_tell = False
+        if isinstance(fx, list):
+            accepted = [a < self._threshold for a in fx]
```

Comment:
Instead of doing this, make sure that fx is a 1d numpy array by using e.g. `fx = pints.vector(fx)`, then use numpy array operations to avoid the for loops

E.g.
```
fx = pints.vector(fx)
accepted = fx < self._threshold   # numpy array of booleans
if np.any(accepted):
    return self._xs[accepted]   # select those elements of xs where accepted == True
else:
    ...
```

I'm also not sure why you'd want to return a None here instead of an empty array? In the other classes we use ``None`` to indicate special occurrences, e.g. a method that requires 3 calls to ask&tell before a sample is generated will return None twice. Here it seems to me that this is a very natural occurrence. So easier for everyone to just return np.array([]) ?

With that, the above is simply

```
fx = pints.vector(fx)
return self._xs[fx < self._threshold]
```